centreon/www/include/configuration/configObject/contact/DB-Func.php

```diff
@@ -102,7 +105,7 @@ function keepOneContactAtLeast($ct_id = null)
 {
     global $pearDB, $form, $centreon;
 
-    if (!isset($contact_id)) {
+    if (!isset($ct_id)) {
```

Comment:
Will fail on next line

Reply:
Hello this fix is good. In fact the $ct_id argument wasn't tested by the isset because before in the isset it was $contact_id but this variable didn't exist :) So this fix is to initialize $contact_id correctly in testing $ct_id by isset()
For example watch this code please, before : 
```
function keepOneContactAtLeast($ct_id = null)
{
    global $pearDB, $form, $centreon;
    if (!isset($contact_id)) {
        $contact_id = $ct_id;
    } elseif (isset($_GET["contact_id"])) {
        $contact_id = htmlentities($_GET["contact_id"], ENT_QUOTES, "UTF-8");
    } else {
        $contact_id = $form->getSubmitValue('contact_id');
    }
```
    
After, with the fix : 
```
function keepOneContactAtLeast($ct_id = null)
{
    global $pearDB, $form, $centreon;
    if (!isset($ct_id)) {
        $contact_id = $ct_id;
    } elseif (isset($_GET["contact_id"])) {
        $contact_id = htmlentities($_GET["contact_id"], ENT_QUOTES, "UTF-8");
    } else {
        $contact_id = $form->getSubmitValue('contact_id');
    }
```

Reply:
That's not what I read 🤔 

Before
```
 if (!isset($contact_id)) {
        $contact_id = $ct_id;
        ...
```
if $contact_id is not set, it will be with $ct_id value (default is null)

After
```
if (!isset($ct_id)) {
        $contact_id = $ct_id;
        ...
```
If $ct_id is not set, you use it to set $contact_id ??
This will result at east in a warning as $ct_id is an undefined variable

![image](https://github.com/user-attachments/assets/31197d8f-f5b8-4ddd-8fbf-7cb90734af53)

Reply:
In fact, this condition will never occur as $ct_id cannot be not set -> by default it will be set to null on line 104
`function keepOneContactAtLeast($ct_id = null)`

Reply:
Sorry I didn't see the ! before isset.
In fact, the code should be like this, with an isset() on $ct_id 😄  
```
function keepOneContactAtLeast($ct_id = null)
{
    global $pearDB, $form, $centreon;
    if (isset($ct_id)) {
        $contact_id = $ct_id;
    } elseif (isset($_GET["contact_id"])) {
        $contact_id = htmlentities($_GET["contact_id"], ENT_QUOTES, "UTF-8");
    } else {
        $contact_id = $form->getSubmitValue('contact_id');
    }
```